CLI: NPM nightly build fails for versions over rc9

The nightly build process is failing at the publish step in NPM, affecting the deployment process.

This is because the script that calculates the next RC version returns an incorrect value if the version is greater than 10 versions.

```
npm notice Publishing to https://registry.npmjs.org/ with tag rc and public access
npm ERR! code E403
npm ERR! 403 403 Forbidden - PUT https://registry.npmjs.org/@dotcms%2fdotcli - You cannot publish over the previously published versions: 1.0.0-rc10.
npm ERR! 403 In most cases, you or one of your dependencies are requesting
npm ERR! 403 a package version that is forbidden by your security policy, or
npm ERR! 403 on a server you do not have access to.

npm ERR! A complete log of this run can be found in: /home/runner/.npm/_logs/2024-05-03T05_18_24_558Z-debug-0.log
Error: Process completed with exit code 1.
```

--- a/.github/actions/publish-npm-cli/action.yml
+++ b/.github/actions/publish-npm-cli/action.yml
@@ -191,7 +191,7 @@ runs:
           echo "LATEST_RC_VERSIONS: ${LATEST_RC_VERSIONS}"
 
           # Extract max RC version
-          MAX_RC_VERSION=$(echo $LATEST_RC_VERSIONS | jq -r --arg filter $VERSION_NPM_FORMAT-rc 'map(.| select(. | contains($filter)) | sub($filter; "") | tonumber ) | $filter+"\(max)"')
+          MAX_RC_VERSION=$(echo $LATEST_RC_VERSIONS | jq -r --arg filter $VERSION_NPM_FORMAT-rc 'map(.| select(. | contains($filter)) | sub($filter; "") | tonumber ) | max')
           echo "MAX_RC_VERSION: ${MAX_RC_VERSION}"
         
           if [[ $MAX_RC_VERSION != null ]]; then

--- a/.github/actions/publish-npm-cli/action.yml
+++ b/.github/actions/publish-npm-cli/action.yml
@@ -193,10 +193,9 @@ runs:
           # Extract max RC version
           MAX_RC_VERSION=$(echo $LATEST_RC_VERSIONS | jq -r --arg filter $VERSION_NPM_FORMAT-rc 'map(.| select(. | contains($filter)) | sub($filter; "") | tonumber ) | $filter+"\(max)"')
           echo "MAX_RC_VERSION: ${MAX_RC_VERSION}"
-      
-          if [[ $MAX_RC_VERSION == "$VERSION_NPM_FORMAT"* ]]; then
-            NEXT_RC_VERSION=$(echo "$MAX_RC_VERSION" | awk -F '-rc' '{print $1 "-rc" $2 + 1}')
-            RC_SUFFIX=$(echo "$NEXT_RC_VERSION" | sed -n 's/.*-rc\([0-9]*\)/-rc\1/p')
+        
+          if [[ $MAX_RC_VERSION != null ]]; then
+            RC_SUFFIX="-rc$(( $MAX_RC_VERSION + 1 ))"
           else
             RC_SUFFIX="-rc1"
           fi;

--- a/.github/actions/publish-npm-cli/action.yml
+++ b/.github/actions/publish-npm-cli/action.yml
@@ -186,13 +186,16 @@ runs:
           fi
 
           echo "VERSION_NPM_FORMAT: ${VERSION_NPM_FORMAT}"
-          LAST_RC_VERSIONS=$(npm view $PACKAGE_FULL_NAME versions --json)
-          echo "LAST_RC_VERSIONS: ${LAST_RC_VERSIONS}"
+          # Get last RC versions
+          LATEST_RC_VERSIONS=$(npm view $PACKAGE_FULL_NAME versions --json)
+          echo "LATEST_RC_VERSIONS: ${LATEST_RC_VERSIONS}"
 
-          LAST_RC_VERSION=$(npm view $PACKAGE_FULL_NAME versions --json | jq --arg filter $VERSION_NPM_FORMAT 'map(.| select(. | contains($filter)))' | jq -r 'map(select(test("-rc\\d+$"))) | max')
-
-          if [[ $LAST_RC_VERSION == "$VERSION_NPM_FORMAT"* ]]; then
-            NEXT_RC_VERSION=$(echo "$LAST_RC_VERSION" | awk -F '-rc' '{print $1 "-rc" $2 + 1}')
+          # Extract max RC version
+          MAX_RC_VERSION=$(echo $LATEST_RC_VERSIONS | jq -r --arg filter $VERSION_NPM_FORMAT-rc 'map(.| select(. | contains($filter)) | sub($filter; "") | tonumber ) | $filter+"\(max)"')
+          echo "MAX_RC_VERSION: ${MAX_RC_VERSION}"
+      
+          if [[ $MAX_RC_VERSION == "$VERSION_NPM_FORMAT"* ]]; then
+            NEXT_RC_VERSION=$(echo "$MAX_RC_VERSION" | awk -F '-rc' '{print $1 "-rc" $2 + 1}')
             RC_SUFFIX=$(echo "$NEXT_RC_VERSION" | sed -n 's/.*-rc\([0-9]*\)/-rc\1/p')
           else
             RC_SUFFIX="-rc1"

--- a/.github/workflows/cli-release-process.yml
+++ b/.github/workflows/cli-release-process.yml
@@ -397,7 +397,7 @@ jobs:
             echo "LATEST_RC_VERSIONS: ${LATEST_RC_VERSIONS}"
 
             # Extract max RC version
-            MAX_RC_VERSION=$(echo $LATEST_RC_VERSIONS | jq -r --arg filter $VERSION_NPM_FORMAT-rc 'map(.| select(. | contains($filter)) | sub($filter; "") | tonumber ) | $filter+"\(max)"')
+            MAX_RC_VERSION=$(echo $LATEST_RC_VERSIONS | jq -r --arg filter $VERSION_NPM_FORMAT-rc 'map(.| select(. | contains($filter)) | sub($filter; "") | tonumber ) | max')
             echo "MAX_RC_VERSION: ${MAX_RC_VERSION}"
           
             if [[ $MAX_RC_VERSION != null ]]; then

--- a/.github/workflows/cli-release-process.yml
+++ b/.github/workflows/cli-release-process.yml
@@ -400,9 +400,8 @@ jobs:
             MAX_RC_VERSION=$(echo $LATEST_RC_VERSIONS | jq -r --arg filter $VERSION_NPM_FORMAT-rc 'map(.| select(. | contains($filter)) | sub($filter; "") | tonumber ) | $filter+"\(max)"')
             echo "MAX_RC_VERSION: ${MAX_RC_VERSION}"
           
-            if [[ $MAX_RC_VERSION == "$VERSION_NPM_FORMAT"* ]]; then
-              NEXT_RC_VERSION=$(echo "$MAX_RC_VERSION" | awk -F '-rc' '{print $1 "-rc" $2 + 1}')
-              RC_SUFFIX=$(echo "$NEXT_RC_VERSION" | sed -n 's/.*-rc\([0-9]*\)/-rc\1/p')
+            if [[ $MAX_RC_VERSION != null ]]; then
+              RC_SUFFIX="-rc$(( $MAX_RC_VERSION + 1 ))"
             else
               RC_SUFFIX="-rc1"
             fi;

--- a/.github/workflows/cli-release-process.yml
+++ b/.github/workflows/cli-release-process.yml
@@ -391,10 +391,17 @@ jobs:
               exit 1
             fi
 
-            LAST_RC_VERSION=$(npm view $PACKAGE_FULL_NAME versions --json | jq --arg filter $VERSION_NPM_FORMAT 'map(.| select(. | contains($filter)))' | jq -r 'map(select(test("-rc\\d+$"))) | max')
+            echo "VERSION_NPM_FORMAT: ${VERSION_NPM_FORMAT}"
+            # Get last RC versions
+            LATEST_RC_VERSIONS=$(npm view $PACKAGE_FULL_NAME versions --json)
+            echo "LATEST_RC_VERSIONS: ${LATEST_RC_VERSIONS}"
 
-            if [[ $LAST_RC_VERSION == "$VERSION_NPM_FORMAT"* ]]; then
-              NEXT_RC_VERSION=$(echo "$LAST_RC_VERSION" | awk -F '-rc' '{print $1 "-rc" $2 + 1}')
+            # Extract max RC version
+            MAX_RC_VERSION=$(echo $LATEST_RC_VERSIONS | jq -r --arg filter $VERSION_NPM_FORMAT-rc 'map(.| select(. | contains($filter)) | sub($filter; "") | tonumber ) | $filter+"\(max)"')
+            echo "MAX_RC_VERSION: ${MAX_RC_VERSION}"
+          
+            if [[ $MAX_RC_VERSION == "$VERSION_NPM_FORMAT"* ]]; then
+              NEXT_RC_VERSION=$(echo "$MAX_RC_VERSION" | awk -F '-rc' '{print $1 "-rc" $2 + 1}')
               RC_SUFFIX=$(echo "$NEXT_RC_VERSION" | sed -n 's/.*-rc\([0-9]*\)/-rc\1/p')
             else
               RC_SUFFIX="-rc1"

